hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieLogFormatWriter.java

```diff
@@ -94,7 +94,8 @@ private FSDataOutputStream getOutputStream() throws IOException, InterruptedExce
       Path path = logFile.getPath();
       if (fs.exists(path)) {
         boolean isAppendSupported = StorageSchemes.isAppendSupported(fs.getScheme());
-        if (isAppendSupported) {
+        boolean needRollOverToNewFile = fs.getFileStatus(path).getLen() > sizeThreshold;
+        if (isAppendSupported && !needRollOverToNewFile) {
```

Comment:
Should we modify it here ? `HoodieAppendHandle#canWrite` already take this ?

Reply:
hi danny， flushToDiskIfRequired cannot work very well in some situation。
in our product env， the log size will roll up to 30g+

Reply:
> hi danny， flushToDiskIfRequired cannot work very well in some situation。 in our product env， the log size will roll up to 30g+

I also encountered a log written to 900G here.

Reply:
The max data block size and log file size are configured by `hoodie.logfile.data.block.max.size` (256MB) and `hoodie.logfile.max.size` (1GB) respectively.  And the rollover based on the size already happens in `appendBlocks`.  Can you check that logic?  not sure if the change here is needed.

Reply:
@XuQianJin-Stars : any updates on this end.